### ISSUE TITLE
[🐛]: tfc heat tag not being assigned when produced by gregtech machines

### ISSUE THREAD

**Issue by PhilPotHT** [TerraFirmaGreg-Team/Modpack-Modern#626]
### Was this bug reported before?

- [X] No, it was not.

### Version modpack used

0.7.14

### Environment

Singleplayer

### New Worlds

Yes

### Please provide a link to your latest.log, if relevant.

https://api.mclo.gs/1/raw/nwvabw8

### Expected Behavior

Items to stack in the same slot with the same serial number in ME system
![Screenshot 2024-12-14 094623](https://github.com/user-attachments/assets/c211752b-7e7f-4830-b8f2-944ae5458019)

### Actual Behavior

Items with tfc heat tag produced by gregtech machines do not have the tag assigned to them so they are not recognised by pattern providers export buses etc. this can be partially mitigated with fuzzy cards but still makes setting up an autocrafting system a complete nightmare.
![Screenshot 2024-12-14 102349](https://github.com/user-attachments/assets/6a7cdecf-f484-472b-8764-cc83ed5df400)

### Steps to Reproduce

Deposit item from inventory and have one processed by machine and directly stored in me system they will be in two separate stack with two different serial numbers
![Screenshot 2024-12-14 130413](https://github.com/user-attachments/assets/076dee00-afd1-4258-8b9e-0fdc91a399bb)
![Screenshot 2024-12-14 130425](https://github.com/user-attachments/assets/0df0834e-a53f-41ca-aa77-5d1c0560b630)
![Screenshot 2024-12-14 130450](https://github.com/user-attachments/assets/3ef0b732-e893-4ac3-b2c9-37c109ed828b)
one is the gregtech super chest with a storage bus the other are processed with a macerator and collected via input bus, they need to be withdrawn and re input to stack in the chest and then not recognised by autocrafting patterns.

### Additional Information

I believe the issue is not that the items have the tag it's that they don't when produced by a machine until picked up by the player, I have searched for the past 3 days to try and find a solution and have come up dry on everything. Really enjoying the modpack but this is a run ender.

**Comment by Pyritie** [comment]
You can work around this by making your patterns with _real_ tfc meltable items as the inputs, and letting JEI fill the outputs, but it's very tedious and frustrating. I store all my dusts in super chests and ME drives if it makes a difference.

**Comment by PhilPotHT** [comment]
Use JEI for the output but manually create the input? - It's extremely frustrating, how are you managing the dusts in chests, redstone in particular I can't seem to pull for any crafting automations

**Comment by Pyritie** [comment]
[2024-12-14 15-42-47.webm](https://github.com/user-attachments/assets/c3cc4849-45e5-4198-87b8-94de3eb16de9)

Here's a video of how to work around the problem, using red alloy ingots as an example.
Note that after making this pattern, you'll have to do the same thing using a "real" red alloy ingot to teach your wiremill how to turn it into wire correctly.
Even after this, sometimes my crafting gets stuck and I'll have to cancel and restart the order.

**Comment by PhilPotHT** [comment]
Thank you the video was a helpful workaround, I can't get it to work with iron dust, everything else (so far) is co operating but iron dust is a firm no from the ae system lol

**Comment by PhilPotHT** [comment]
This is becoming more of an issue in storage than it is autocrafting, the super chests really don't want to work with some things, I'm using a fuzzy card with the physical one in the tag, tried dragging the tag in from emi and still no luck

**Comment by Pyritie** [comment]
> Thank you the video was a helpful workaround, I can't get it to work with iron dust, everything else (so far) is co operating but iron dust is a firm no from the ae system lol

yeah, I've had to start and stop recipes a few time but at least it's _usable_, my setup constantly has trouble with wrought iron and steel plates lol

**Comment by PhilPotHT** [comment]
Have you managed to get the system to store all the dusts in one chest? I've sorted the autocrafting issue (for now) but I still can't get the dusts to stack in the gregtech chests

**Comment by Pyritie** [comment]
No, I use super chests/tanks for things I have tons of, partitioned ME drives for other dusts and ingots, and a giant item vault for everything else

**Comment by PhilPotHT** [comment]
Didn’t think about an item vault, I’ve more or less got them stored with the Greg tech out variant, just can’t deposit any dusts directly into the me system but it is working so far (touch wood) just takes trial and error for the crafting patterns on which input/output to use. Such a frustrating bug spent the past 2 play sessions just trying to troubleshoot and organise storage

**Comment by PhilPotHT** [comment]
Had the exact same idea, having the non tagged one in the chest seems to be working out alright

**Comment by Fomys** [comment]
Maybe this can work too for the issue https://github.com/TerraFirmaGreg-Team/Modpack-Modern/issues/543#issuecomment-2557743065

**Comment by Pyritie** [comment]
Great find! We're currently working to update gtceu to their latest release (since TFG is so far behind), so if your change is accepted into gtceu, we'll get your update at some point in the future (possibly gtceu 1.7?)
Otherwise, if they don't accept it, TFG-Core is open source now, so you could submit a PR into that instead?

**Comment by jojo7682345** [comment]
I don't really know how to fix the bug with code in another mod as the bug is deeply embedded within gtceu. For now the only solution I see is to clone the repo, checkout the version used in TFG, (which I think 1.20.1-1.2.3.a-build_771) and apply the changes I made in GregTechCEu/GregTech-Modern#2951. Although this isn't very elegant.

**Comment by Pyritie** [comment]
let's see if it gets accepted into gtceu, but if not, I think a mixin might be able to do it? I've not used them before but from what I understand they're a bit of code that can modify stuff inside another mod. They are pretty frail though so if gtceu gets the "proper" fix then I would much prefer that

**Comment by Pyritie** [comment]
(btw if you want to talk to some of the other contributors, we have a thread in the TFG discord [here](https://discord.com/channels/400913133620822016/1331744604701069393)!)

**Comment by jojo7682345** [comment]
The fix is denied by gtceu, however I have made a mixin to fix the issue and submitted it to the TFG-Core mod.
TerraFirmaGreg-Team/Core-Modern#18
As pointed out by a member of the GTCEu team my first solution was not completely adressing the root cause of the issue. However fixing it completely is infeasible without altering both gtceu and ae2. The mixin should fix the issue for people using the TFC mod in combination with the other two, but isn't a propper solution to implement in either one of the mods.
